vyper/semantics/types/module.py

```diff
@@ -45,28 +45,35 @@ def __init__(
         functions: dict,
         events: dict,
         structs: dict,
+        constants: Optional[dict] = None,
     ) -> None:
         validate_unique_method_ids(list(functions.values()))
 
-        members = functions | events | structs
+        constants = constants if constants else {}
```

Comment:
`constants or {}`

Reply:
tbh we should do this for all the other things too, since in most cases we end up passing an empty dict for most of these parameters

Reply:
do you mean to set the default for these params to None? should I do it in this PR?

Reply:
Yes and yes!

Reply:
I am thinking that since `_from_lists` is already passing a dictionary (either empty or with values) for all the params, and it is the only caller of the ctor for `InterfaceT`, maybe we can require a dict instead for all the same params in the ctor for `InterfaceT`?

Reply:
I made the changes in https://github.com/vyperlang/vyper/pull/4253/commits/69e5b3e9bc38426e68aafc8c9463083ce1de77c3 to show what I meant in the previous comment.